src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -167,25 +184,32 @@ private bool TrimMethods(ITaskItem assemblyItem)
         if (isTrimmed)
         {
             AddItemToTrimmedList(assemblyFilePathArg, trimmedAssemblyFilePath);
+            newAssmeblyItem.ItemSpec = trimmedAssemblyFilePath;
         }
 
+        _updatedAssemblies.Add(newAssmeblyItem);
```

Comment:
Instead of another item, you can make `UpdatedAssemblies` the only output item. And set a metadata on the item like `ILStripped=true`. This way it is a clean output. And if caller does need to differentiate somehow then they have a simple way of doing that.
Also, in case of a trimmed assembly add `UntrimmedAssemblyPath` metadata to the item. This will help in any debugging, or if the caller needs the original path too.

Reply:
The name of the file path should indicate it is a trimmed assembly.

Reply:
The targets might need to differentiate. It is a general convention in msbuild to add such metadata which can help with filtering. Otherwise, any targets would need to assume the path structure.

Reply:
Updated.